packages/block-editor/src/hooks/dimensions.js

```diff
@@ -131,6 +135,48 @@ function DimensionsPanelPure( { clientId, name, setAttributes, settings } ) {
 // and not the whole attributes object.
 export const DimensionsPanel = pure( DimensionsPanelPure );
 
+/**
+ * Determine whether there is block support for dimensions.
+ *
+ * @param {string} blockName Block name.
+ * @param {string} feature   Background image feature to check for.
+ *
+ * @return {boolean} Whether there is support.
+ */
+export function hasDimensionsSupport( blockName, feature = 'any' ) {
+	if ( Platform.OS !== 'web' ) {
+		return false;
+	}
+
+	const support = getBlockSupport( blockName, DIMENSIONS_SUPPORT_KEY );
+
+	if ( support === true ) {
+		return true;
+	}
+
+	if ( feature === 'any' ) {
+		return !! support?.dimensions;
+	}
+
+	return !! support?.[ feature ];
+}
+
+export default {
+	useBlockProps,
+	attributeKeys: [ 'style' ],
+	hasSupport( name ) {
+		return hasDimensionsSupport( name, 'aspectRatio' );
+	},
+};
+
+function useBlockProps( { style } ) {
+	const className = classnames( {
+		'has-aspect-ratio': !! style?.dimensions?.aspectRatio,
+	} );
+
+	return { className };
```

Comment:
TODO: Add more checks before output here. I've left this pretty bare for the moment while I debug why `has-aspect-ratio` is only output some of the time 🤔